slycot/transform.py

```diff
@@ -333,6 +333,254 @@ def tb04ad(n,m,p,A,B,C,D,tol1=0.0,tol2=0.0,ldwork=None):
+             response matrix G(freq).
+
+      hinvb : complex p-by-m array which contains the
+                      -1
+             product H  B.
+
+
+    Raises
+    ------
+      ValueError : e
+        e.info contains information about the exact type of exception.
+         < 0 : if info = -i, the ith argument had an illegal value;
+         = 1 : More than 30 iterations were required to isolate the
+               eigenvalues of A. The computation is continued ?.
+         = 2 : Either FREQ is too near to an eigenvalue of A, or RCOND
+               is less than the machine precision EPS.
+
+    Example
+    -------
+    >>> A = np.array([[0.0, 1.0],
+                [-100.0,   -20.1]])
+    >>> B = np.array([[0.],[100]])
+    >>> C = np.array([[1., 0.]])
+    >>> n = np.shape(A)[0]
+    >>> m = np.shape(B)[1]
+    >>> p = np.shape(C)[0]
+    >>> jw_s = [1j*10, 1j*15]
+    >>> at, bt, ct, g_1, hinvb, info = slycot.tb05ad(n, m, p, jw_s[0],
+                                                    A, B, C, job='NG')
+    >>> g_2, hinv2,info = slycot.tb05ad(n, m, p, jw_s[1], at, bt, ct, job='NH')
+
+    """
+    def error_handler(out, arg_list):
+        if out[-1] < 0:
+            # Conform fortran 1-based argument indexing to
+            # to python zero indexing.
+            error_text = ("The following argument had an illegal value: "
+                          + arg_list[-out[-1]-1])
+            e = ValueError(error_text)
+            e.info = out[-1]
+            raise e
+        if out[-1] == 1:
+            error_text = ("More than 30*"+str(n)+"iterations are required "
+                          "to iso late the eigenvalue of A; the computations "
+                          "are continued.")
+            e = ValueError(error_text)
+            e.info = out[-1]
+            raise e
+        if out[-1] == 2:
+            error_text = ("Either FREQ is too near to an eigenvalue of A, or "
+                          "RCOND is less than the machine precision EPS.")
+            e = ValueError(error_text)
+            e.info = out[-1]
+            raise e
+
+    hidden = ' (hidden by the wrapper)'
+    arg_list = ['baleig'+hidden, 'inita'+hidden, 'n', 'm', 'p', 'freq', 'a',
+                'lda'+hidden, 'b', 'ldb'+hidden, 'c', 'ldc'+hidden, 'rcond',
+                'g', 'ldg'+hidden, 'evre', 'evim', 'hinvb', 'ldhinv'+hidden,
+                'iwork'+hidden, 'dwork'+hidden, 'ldwork'+hidden,
+                'zwork'+hidden, 'lzwork'+hidden, 'info'+hidden]
+    # Fortran function prototype:
+    # TB05AD(baleig,inita,n,m,p,freq,a,lda,b,ldb,c,ldc,rcond,g,ldg,evre,evim,hinvb,ldhinv,
+    # iwork,dwork,ldwork,zwork,lzwork,info)
+
+    # Sanity check on matrix dimensions
+    if A.shape != (n, n):
+        e = ValueError("The shape of A is (" + str(A.shape[0]) + "," +
+                       str(A.shape[1]) + "), but expected (" + str(n) +
+                       "," + str(n) + ")")
+        e.info = -6  
```

Comment:
I'm not sure it's worth faking the `info` attribute.  You haven't done it for the "Unrecognized job" error at the end (where it's not clear what it would be...).